app/controllers/common/FileUploadHelper.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.common
+
+import javax.inject.Inject
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import play.api.Configuration
+import play.api.i18n.{I18nSupport, Messages, MessagesApi}
+import play.api.mvc.{MessagesControllerComponents, RequestHeader, Result}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import models.{DraftId, Mode, NormalMode, UploadedFile, UserAnswers}
+import navigation.Navigator
+import pages.{Page, QuestionPage}
+import services.fileupload.FileService
+import views.html.{UploadLetterOfAuthorityView, UploadSupportingDocumentsView}
+
+case class FileUploadHelper @Inject() (
+  override val messagesApi: MessagesApi,
+  override val controllerComponents: MessagesControllerComponents,
+  supportingDocumentsView: UploadSupportingDocumentsView,
+  letterOfAuthorityView: UploadLetterOfAuthorityView,
+  fileService: FileService,
+  navigator: Navigator,
+  configuration: Configuration
+)(implicit ec: ExecutionContext)
+    extends I18nSupport
+    with FrontendBaseController {
+
+  private val maxFileSize: Long = configuration.underlying.getBytes("upscan.maxFileSize") / 1000000L
+
+  def showPage(mode: Mode, draftId: DraftId, isLetterOfAuthority: Boolean)(implicit
+    request: RequestHeader
+  ): Future[Result] = {
+    val redirectPath = getRedirectPath(draftId, isLetterOfAuthority, mode)
+    fileService.initiate(draftId, redirectPath, isLetterOfAuthority).map {
+      response =>
+        Ok(
+          supportingDocumentsView(
+            draftId = draftId,
+            upscanInitiateResponse = Some(response),
+            errorMessage = None
+          )
+        )
```

Comment:
Will this need to be wrapped in a boolean check (isLetterOfAuthority to allow for the letter of authority view when that story is done? I'm thinking along the lines of the showErrorPage method.

Reply:
Answer: yes.